### PR TITLE
100G lr fr dr support

### PR DESCRIPTION
**Pull request by vjanandr**
SFF changes to support 100G BASE LR, FR and DR.

### CODE REVIEW

File: modules/sff/module/inc/sff/sff.x [hunk 1]
```diff
@@ -127,6 +130,9 @@ SFF_MODULE_TYPE_ENTRY(1G_BASE_CX, 1GBASE-CX)
 SFF_MODULE_TYPE_ENTRY(1G_BASE_T, 1GBASE-T)
 SFF_MODULE_TYPE_ENTRY(100_BASE_LX, 100BASE-LX)
 SFF_MODULE_TYPE_ENTRY(100_BASE_FX, 100BASE-FX)
+SFF_MODULE_TYPE_ENTRY(100G_BASE_LR, 100GBASE-LR)
```
Comment (ashutosh-arrcus) [inline modules/sff/module/inc/sff/sff.x]: You should group the 100G enums together.

File: modules/sff/module/inc/sff/sff.x [hunk 2]
```diff
@@ -59,6 +59,9 @@ SFF_MEDIA_TYPE_ENTRY(1G_BASE_CX, 1GBASE-CX)
 SFF_MEDIA_TYPE_ENTRY(1G_BASE_T, 1GBASE-T)
 SFF_MEDIA_TYPE_ENTRY(100_BASE_LX, 100BASE-LX)
 SFF_MEDIA_TYPE_ENTRY(100_BASE_FX, 100BASE-FX)
+SFF_MEDIA_TYPE_ENTRY(100G_BASE_LR, 100GBASE-LR)
```
Comment (ashutosh-arrcus) [inline modules/sff/module/inc/sff/sff.x]: You should group the 100G enums together.

File: modules/sff/module/python/sff/enums.py [hunk 3]
```diff
@@ -213,7 +213,10 @@ class SFF_MODULE_TYPE(Enumeration):
     _1G_BASE_T = 49
     _100_BASE_LX = 50
     _100_BASE_FX = 51
-    _4X_MUX = 52
+    _100G_BASE_LR = 52
```
Comment (ashutosh-arrcus) [inline modules/sff/module/python/sff/enums.py]: You should group the 100G enums together.

File: modules/sff/module/src/sff.c [hunk 4]
```diff
@@ -357,6 +372,9 @@ sff_media_type_get(sff_module_type_t mt)
         case SFF_MODULE_TYPE_1G_BASE_ZX:
         case SFF_MODULE_TYPE_100_BASE_LX:
         case SFF_MODULE_TYPE_100_BASE_FX:
+        case SFF_MODULE_TYPE_100G_BASE_LR:
```
Comment (ashutosh-arrcus) [inline modules/sff/module/src/sff.c]: You should group the 100G enums together.

File: modules/sff/module/src/sff_enums.c [hunk 5]
```diff
@@ -847,6 +847,9 @@ aim_map_si_t sff_module_type_map[] =
     { "1G_BASE_T", SFF_MODULE_TYPE_1G_BASE_T },
     { "100_BASE_LX", SFF_MODULE_TYPE_100_BASE_LX },
     { "100_BASE_FX", SFF_MODULE_TYPE_100_BASE_FX },
+    { "100G_BASE_LR", SFF_MODULE_TYPE_100G_BASE_LR },
```
Comment (ashutosh-arrcus) [inline modules/sff/module/src/sff_enums.c]: You should group the 100G enums together.

File: modules/sff/module/src/sff_enums.c [hunk 6]
```diff
@@ -905,6 +908,9 @@ aim_map_si_t sff_module_type_desc_map[] =
     { "1GBASE-T", SFF_MODULE_TYPE_1G_BASE_T },
     { "100BASE-LX", SFF_MODULE_TYPE_100_BASE_LX },
     { "100BASE-FX", SFF_MODULE_TYPE_100_BASE_FX },
+    { "100GBASE-LR", SFF_MODULE_TYPE_100G_BASE_LR},
```
Comment (ashutosh-arrcus) [inline modules/sff/module/src/sff_enums.c]: You should group the 100G enums together.

File: modules/sff/module/inc/sff/8636.h [hunk 7]
```diff
@@ -185,6 +190,17 @@
 #define SFF8636_MEDIA_25GE_CR_N(idprom)                  \
   (idprom[192] == SFF8636_CC192_25GE_CR_N)
 
+#define SFF8636_MEDIA_100GE_LR(idprom)                   \
```
Comment (ashutosh-arrcus) [inline modules/sff/module/inc/sff/8636.h]: You should group the 100G enums together.

File: modules/sff/module/auto/sff.yml [hunk 8]
```diff
@@ -148,6 +148,12 @@ sff_module_types: &sff_module_types
     desc: "100BASE-LX"
 - 100_BASE_FX:
     desc: "100BASE-FX"
+- 100G_BASE_LR:
```
Comment (ashutosh-arrcus) [inline modules/sff/module/auto/sff.yml]: You should group the 100G enums together.

File: modules/sff/module/inc/sff/sff.x (hunk 1 above)
Comment (vijayr-arrcus) [reply modules/sff/module/inc/sff/sff.x]: grouped 100G...

File: modules/sff/module/inc/sff/sff.x (hunk 2 above)
Comment (vijayr-arrcus) [reply modules/sff/module/inc/sff/sff.x]: grouped 100G...

File: modules/sff/module/python/sff/enums.py (hunk 3 above)
Comment (vijayr-arrcus) [reply modules/sff/module/python/sff/enums.py]: grouped 100G...

File: modules/sff/module/src/sff.c (hunk 4 above)
Comment (vijayr-arrcus) [reply modules/sff/module/src/sff.c]: grouped 100G...

File: modules/sff/module/src/sff_enums.c (hunk 6 above)
Comment (vijayr-arrcus) [reply modules/sff/module/src/sff_enums.c]: grouped 100G...

File: modules/sff/module/inc/sff/8636.h (hunk 7 above)
Comment (vijayr-arrcus) [reply modules/sff/module/inc/sff/8636.h]: grouped 100G...

File: modules/sff/module/auto/sff.yml (hunk 8 above)
Comment (vijayr-arrcus) [reply modules/sff/module/auto/sff.yml]: grouped 100G...

File: modules/sff/module/inc/sff/8636.h [hunk 9]
```diff
@@ -149,6 +149,11 @@
 #define SFF8636_CC192_40GE_SWDM4         0x1F
 #define SFF8636_CC192_100GE_SWDM4        0x20
 #define SFF8636_CC192_100GE_PAM4_BIDI    0x21
+#define SFF8636_CC192_100GE_LR           0x27
```
Comment (ashutosh-arrcus) [inline modules/sff/module/inc/sff/8636.h]: Please define them in increasing order of their value.

Comment (vijayr-arrcus) [reply modules/sff/module/inc/sff/8636.h]: addressed..

**Review by ashutosh-arrcus — APPROVED** [review]

**Pull request closed without merging**